api/types/installer.go

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package types
+
+import (
+	"time"
+
+	"github.com/gravitational/trace"
+)
+
+// Installer is an installer script rseource
+type Installer interface {
+	Resource
+
+	// GetScript returns the contents of the installer script
+	GetScript() string
+	// SetScript sets the installer script
+	SetScript(string)
+
+	String() string
+}
+
+// NewInstallerV1 returns a new installer resource
+func NewInstallerV1(script string) (*InstallerV1, error) {
+	installer := &InstallerV1{
+		Metadata: Metadata{},
+		Spec: InstallerSpecV1{
+			Script: script,
+		},
+	}
+	if err := installer.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return installer, nil
+}
+
+func MustNewInstallerV1(script string) *InstallerV1 {
```

Comment:
```suggestion
// MustNewInstallerV1 creates a new installer resource from the provided script.
//
// Panics in case of any error when creating the resource.
func MustNewInstallerV1(script string) *InstallerV1 {
```